plugins/woocommerce/src/Admin/Features/ProductBlockEditor/ProductTemplates/BaseProductTemplate.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Product Block Editor reusable template pieces.
+ */
+
+namespace Automattic\WooCommerce\Admin\Features\ProductBlockEditor\ProductTemplates;
+
+trait BaseProductTemplate {
+
+    /**
+     * Add all base template groups.
+     */
+    protected function add_base_template() {
+        $this->add_general_group();
+        $this->add_pricing_group();
+    }
+
+    /**
+     * Add general group and fields.
+     */
+    protected function add_general_group() {
+        $this->add_group(
+            array(
+                'id'    => self::GENERAL_GROUP,
+                'title' => __( 'General', 'woocommerce' ),
+                'order' => 10,
```

Comment:
Why is `order` necessary? It would be nice to get rid of these hardcoded orders if possible.

Reply:
The `order` property for tabs is specific to a bug with rendering tabs on the frontend.  See [this PR](https://github.com/woocommerce/woocommerce/pull/38081) for more context.

But in general, I do not think we're moving away from using `order`.  While I originally liked not having an `order` property, this would create too much dependency and require too much knowledge of the existing template.

For example, imagine we had `$template->insert_field_before( 'woocommerce/product-name-field', $my_field )`.  This requires the consumer to know the name `woocommerce/product-name-field`, prevents us from changing to a different block without breaking third party implementations, and if some of our templates did not include this field, the third party field would also not be included.

Reply:
Additionally, it also creates a lot of issues with timing.  If two plugins each insert at the same block, which should render first?

```
// PluginA
$template->insert_field_before( 'woocommerce/product-name-field', $field_a );
// PluginB
$template->insert_field_before( 'woocommerce/product-name-field', $field_b );
```

Whichever was loaded later will be the one adjacent to the name field.  If these were on the same hook, the alphabetical order of the plugins would dictate which function runs first.

```
<ProductNameField /> // Must be added first in order to work
<FieldB /> // Added after A, so inserted next to the product name field
<FieldA /> // Added before B
```

Compare this to including an `order` property which has specific intent and removes the timing issues we see between various hooks or plugin execution order.